core/color/color_test.go

```diff
@@ -0,0 +1,43 @@
+package color
+
+import (
+	"testing"
+)
+
+func TestOpposite(t *testing.T) {
+	testCases := []struct {
+		desc string
+		in   string
+		want string
+	}{
+		{
+			desc: "white=>black",
+			in:   "W",
+			want: "B",
+		},
+		{
+			desc: "black=>white",
+			in:   "B",
+			want: "W",
+		},
+		{
+			desc: "empty=>empty",
+			in:   "",
+			want: "",
+		},
+		{
+			desc: "any=>any",
+			in:   "any",
+			want: "any",
+		},
+	}
+
```

Comment:
The for loop here below effectively iterates through the test struct input and output values and both outputs the test results and a simple string description of each iteration's input and output struct testCases' slice arrays values. The string description facilitates following the logic of this test method, and identifying where failures may occur. Entering bad data on the color.go side causes this test to fail consitently, triggering the 'if out != Color(tc.want) {...}' conditional below, demonstrating that this test is effective.